cmd/manager/main.go

```diff
@@ -105,8 +130,7 @@ func main() {
 		Namespace:      namespace,
 		NgrokAPIDriver: ngrokapidriver.NewNgrokApiClient(ngrokApiKey),
 	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "ingress")
-		os.Exit(1)
+		return fmt.Errorf("unable to create ingress controller: %w", err)
```

Comment:
logging got moved up to `main`, where it just prints the error and exits.
This removes a little boilerplate, but I'm not totally tied to that decision. It does seem slightly cleaner code-wise, but it means we can't easily provide context (key-value logging pair context I mean) to the messages. Still, we could add additional messages if we wanted that.

Reply:
This seems fine to me. we can still add context at the manager level, which probably is most appropriate since its the manager failing to add the new reconciler.